kit/ChildSession.cpp

```diff
@@ -56,7 +56,8 @@ using Poco::URI;
 using namespace LOOLProtocol;
 
 bool ChildSession::NoCapsForKit = false;
-UnoCommandsRecorder ChildSession::unoCommandsRecorder;
+int ChildSession::CurrentPos = 0;
+std::shared_ptr<UnoCommandsRecorder>ChildSession::UnoCommandsRecorders[ChildSession::NUM_UNO_RECORDERS];
```

Comment:
Why did you introduce several UNO command recorders?  Why bother with shared_ptr here? Just more complication, more things that can be wrong. Michael said that we want this to be as minimal as possible.

A Kit processes handles the editing of one document and then dies, there is no need to try to make sure memory is deallocated cleanly after a ChildSession goes away or anything like that.

Reply:
Each child session has its own uno commands recorder which records the uno commands that were fired by the user Tor

Reply:
This output might make it clear

![Image Pasted at 2021-7-23 01-00](https://user-images.githubusercontent.com/56198900/126753638-b44b0af1-d434-4ee1-b05c-d19a6ceb22a3.png)

Reply:
Tor if they are not shared pointers then how will we manage their memory when they are out of scope? 

Example Scenario:

There are 50 recorders recording from 50 different child session. Suppose a new user comes in so the 51st recorder will be replaced at the 0 th position of bookkeeping array.  So what happens to the old recorder at 0th position and how it  will be deallocated? We cannot deallocate it with childsession destructor because the bookeeping array will be holding it as well. (There is no way to detect whether an address was free or not).

Reply:
Using a shared pointer when we can ensure that atleast 1 reference is alive is safe inside a signal handler. We are not allocating or freeing anything. In fact we are not even modifying the control block in any way.

Reply:
In summary what I want to tell is that there is a need for memory management in case of unorecorders or else  it will result in leaking memory

Reply:
![image](https://user-images.githubusercontent.com/56198900/126756111-ef3e44d2-a833-4454-a6cb-26fdc886de85.png)